pkg/resource/endpoint/custom_update_api.go

```diff
@@ -0,0 +1,123 @@
+package endpoint
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	ackcompare "github.com/aws-controllers-k8s/runtime/pkg/compare"
+	ackerr "github.com/aws-controllers-k8s/runtime/pkg/errors"
+	ackrequeue "github.com/aws-controllers-k8s/runtime/pkg/requeue"
+	ackrtlog "github.com/aws-controllers-k8s/runtime/pkg/runtime/log"
+	svcsdk "github.com/aws/aws-sdk-go/service/eventbridge"
+)
+
+// customUpdateEndpoint deals with the eventually consistent API of Endpoints in
+// a generic manner. It will always requeue a resource after modification to
+// sync its spec and status.
+//
+// note: there are several edge cases in this API which we don't cover e.g.,
+// going from a DISABLED replication state to ENABLED with an invalid ARN. The
+// controller currently has no logic to reconcile this against the eventual
+// consistent nature and inconsistent behavior of the API (API replication config will remain
+// DISABLED and state UPDATE_FAILED and does not reflect the desired state)
+func (rm *resourceManager) customUpdateEndpoint(
```

Comment:
From what I can tell, this doesn't need to be a custom update method... you could save yourself some hassle (in maintaining the code) by using the standard generated update code paths with a couple hooks for `sdk_update_pre_build_request` (for the `validateEndpointSpec` and `endpointInMutatingState` calls) and the `sdk_update_post_build_request` hook point for the `unsetRemovedSpecFields` call.

Reply:
Agree, we went down this path due to some API quirks, but then settled on these couple of checks/optimizations. I'll revert to code-gen approach with templates.